src/AppSystem/Launcher.vala

```diff
@@ -328,4 +354,24 @@ public class Dock.Launcher : BaseItem {
             multiple_windows_open = app.windows.length > 1;
         }
     }
+
+    private void on_enter () {
+        if (timeout_id == 0) {
+            timeout_id = Timeout.add_once (POPOVER_NAME_TIMEOUT, show_popover_name);
+        }
```

Comment:
I think we actually don't want a timeout at all. It would fix https://github.com/elementary/dock/issues/29 to show them right away